Rename loggers/base.py to loggers/logger.py and changed LightningLoggerBase to Logger

MohammedAlkhrashi opened the pull request:
## What does this PR do?

<!--
Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change.

If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

The following links the related issue to the PR (https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
-->

Fixes \#11971
- Changed loggers.logger.LightningLoggerBase to loggers.logger.Logger
- Then renamed loggers/base.py to loggers/logger.py.  
- Created a new loggers/base.py, which contains LightningLoggerBase with a deprecation warning and subclasses from Logger to support backward compatibility. 
- Changed all internal imports to support the refactoring. 
- Changed loggers/test_base.py to loggers/test_logger.py 
- Created a new loggers/test_base.py to test the deprecation warning from loggers.base.LightningLoggerBase

I still haven't updated the documentation, I first wanted to make sure that I am on the right track and if there are any necessary changes needed. 

### Does your PR introduce any breaking changes? If yes, please list them.

<!-- FILL IN or None -->

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

<!-- In the CHANGELOG, separate each item in the unreleased section by a blank line to reduce collisions -->

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

- [x] Had fun? 😄

daniellepintz commented:
@MohammedAlkhrashi Thanks so much for your PR! Mind fixing the merge conflicts and failing tests so we can get this merged?

MohammedAlkhrashi commented:
> @MohammedAlkhrashi Thanks so much for your PR! Mind fixing the merge conflicts and failing tests so we can get this merged?

Alright 👍 , I will try to fix the conflicts and the failing tests by next week if that's okay.

daniellepintz commented:
Awesome, thanks!

ananthsub commented on pytorch_lightning/loggers/__init__.py:
```diff
@@ -13,11 +13,14 @@
 from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
 
-__all__ = ["LightningLoggerBase", "LoggerCollection", "TensorBoardLogger", "CSVLogger"]
+__all__ = ["Logger", "LoggerCollection", "TensorBoardLogger", "CSVLogger", "LightningLoggerBase"]
```
Sort these alphabetically

ananthsub commented on pytorch_lightning/loggers/base.py:
```diff
@@ -60,405 +33,9 @@ class LightningLoggerBase(ABC):
         rank_zero_deprecation(
-            "`LoggerCollection.close` method is deprecated in v1.5 and will be removed in v1.7."
-            " Please use `LoggerCollection.finalize` instead."
+            "The `pl.loggers.base.LightningLoggerBase` is deprecated. "
+            " Please use `pl.loggers.logger.Logger` instead."
```
```suggestion
            "`pytorch_lightning.loggers.base.LightningLoggerBase` is deprecated. "
            " Please use `pytorch_lightning.loggers.logger.Logger` instead."
```

akihironitta replied:
In addition, should we note that this deprecated class will be removed in v1.8?

akihironitta reviewed (commented):
Could you also update occurrences in the docs?

```console
$ grep -rnI LightningLoggerBase docs
docs/source/extensions/logging.rst:207:You can implement your own logger by writing a class that inherits from :class:`~pytorch_lightning.loggers.base.LightningLoggerBase`.
docs/source/extensions/logging.rst:212:    from pytorch_lightning.loggers.base import LightningLoggerBase, rank_zero_experiment
docs/source/extensions/logging.rst:216:    class MyLogger(LightningLoggerBase):
docs/source/common/lightning_module.rst:996:        # List of LightningLoggerBase objects
docs/source/common/trainer.rst:1755:    # List of LightningLoggerBase objects
```

MohammedAlkhrashi commented:
Hi everyone, I've fixed the merge conflicts, failed tests, and updated the docs accordingly.  However, I was wondering if we should also implement wrappers for other functionality in current loggers/base.py for backward compatibility? Such as DummyLogger, or LoggerCollection. Currently, only LightiningLoggerBase is backward compatible if imported directly from 'pl.loggers.base'.

Additionally, I am not sure about the depreciation test procedure exactly, should I move my deprecation tests to tests/deprecated_api?

awaelchli commented on tests/loggers/test_base.py:
```diff
@@ -139,259 +64,6 @@ def after_save_checkpoint(self, checkpoint_callback):
-        trainer.fit(model, dm)
+def test_lightning_logger_base_deprecation_warning():
+    with pytest.deprecated_call(match="The `pytorch_lightning.loggers.base.LightningLoggerBase` is deprecated."):
+        CustomDeprecatedLogger()
```
can we move this test to the file `tests/deprecated_api/test_remove_1-9.py`. Thanks :)

awaelchli commented:
Great work!

> I was wondering if we should also implement wrappers for other functionality in current loggers/base.py for backward compatibility? Such as DummyLogger, or LoggerCollection. 

Yes exactly, we should. The exception is LoggerCollection where you would need to just keep the existing deprecation message (because it gets removed in 1.8).

> Additionally, I am not sure about the deprecation test procedure exactly, should I move my deprecation tests to tests/deprecated_api? If so, I am assuming it's moved to the 1.8 files, is that correct?

Yes, see my latest commit and the addition in `test_remove_1-9.py`. The same would be done for the other classes that had to move.

MohammedAlkhrashi commented:
> Yes exactly, we should. The exception is LoggerCollection where you would need to just keep the existing deprecation message (because it gets removed in 1.8).
> 
Great, just to make sure, this would include creating wrappers for all classes and functions including zero_rank_experiment and merge_dicts, right?
As for the LoggerCollection, that means I would still create a wrapper for it as well, however, I shouldn't create a new deprecation test as one already exists, did I get this correctly?
>Yes, see my latest commit and the addition in test_remove_1-9.py. The same would be done for the other classes that had to move.
Got it, thanks @awaelchli 👍

awaelchli commented:
Yes, I think so. I see no other way :)

akihironitta commented on CHANGELOG.md:
```diff
@@ -44,7 +44,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Deprecated
 
--
+- Deprecated `loggers.base.LightningLoggerBase` in favor of `loggers.logger.Logger`, and deprecated `loggers.base.py` in favor of `loggers.logger.py` ([#120148](https://github.com/PyTorchLightning/pytorch-lightning/pull/12014))
```
I prefer it to be explicit :)
```suggestion
- Deprecated `pytorch_lightning.loggers.base.LightningLoggerBase` in favor of `pytorch_lightning.loggers.logger.Logger`, and deprecated `pytorch_lightning.loggers.base` in favor of `pytorch_lightning.loggers.logger` ([#120148](https://github.com/PyTorchLightning/pytorch-lightning/pull/12014))
```

akihironitta reviewed (approved):
Thank you for your contribution!
